compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -1837,15 +1840,19 @@ object Types {
       case mt: MethodType if !mt.isParamDependent =>
         val formals1 = if (dropLast == 0) mt.paramInfos else mt.paramInfos dropRight dropLast
         val isContextual = mt.isContextualMethod && !ctx.erasedTypes
-        val isErased = mt.isErasedMethod && !ctx.erasedTypes
+        val isErased =
+          (if mt.isErasedMethod && !ctx.erasedTypes then
+            mt.erasedParams
+          else List.fill(mt.paramInfos.size) { false }) dropRight dropLast
```

Comment:
```suggestion
        val isErased = mt.erasedParams.dropRight(dropLast)
```

`erasedParams` should be defined in `MethodType`.